apstools/utils.py

```diff
@@ -2325,3 +2327,203 @@ def copy_filtered_catalog(source_cat, target_cat, query=None):
+                    )
+            elif isinstance(v, ophyd.Device):
+                # print("Device", v.name)
+                self._device_name = _nm_base + [k]
+                if v.name not in self._known_device_names:
+                    self._odb[v.name] = ".".join(self._device_name)
+                    self._known_device_names.append(v.name)
+                    self._ophyd_epicsobject_walker(v)
+
+    def _ref_dict(self, parent, key):
+        """Accessor used by ``_ophyd_epicsobject_walker()``"""
+        return parent[key]
+
+    def _ref_object_attribute(self, parent, key):
+        """Accessor used by ``_ophyd_epicsobject_walker()``"""
+        return getattr(parent, key)
+
+    def _register_signal(self, signal, pv, mode):
+        """Register a signal with the given mode."""
+        fdn = full_dotted_name(signal)
+        if fdn not in self._pvdb[pv][mode]:
+            self._pvdb[pv][mode].append(fdn)
+
+    def _signal_processor(self, signal):
+        """Register a signal's read & write PVs."""
+        self._register_signal(signal, signal._read_pv.pvname, "R")
+        if hasattr(signal, "_write_pv"):
+            self._register_signal(signal, signal._write_pv.pvname, "W")
+
+    def search_by_mode(self, pvname, mode="R"):
+        """Search for PV in specified mode."""
+        if mode not in ["R", "W"]:
+            raise ValueError(
+                f"Incorrect mode given ({mode}." "  Must be either `R` or `W`."
+            )
+        return self._pvdb[pvname][mode]
+
+    def search(self, pvname):
+        """Search for PV in both read & write modes."""
+        return dict(
+            read=self.search_by_mode(pvname, "R"),
+            write=self.search_by_mode(pvname, "W"),
+        )
+
+    def ophyd_search(self, oname):
+        """Search for ophyd object by ophyd name."""
+        return self._odb.get(oname)
+
+
+def _get_pv_registry(force_rebuild, ns):
+    """
+    Check if need to build/rebuild the PV registry.
+
+    PARAMETERS
+
+    force_rebuild
+        *bool* :
+        If ``True``, rebuild the internal registry that maps
+        EPICS PV names to ophyd objects.
+    ns
+        *dict* or `None` :
+        Namespace dictionary of Python objects.
+
+    """
+    global _findpv_registry
+    if _findpv_registry is None or force_rebuild:
+        _findpv_registry = PVRegistry(ns=ns)
+    return _findpv_registry
+
+
+def findname(oname, force_rebuild=False, ns=None):
+    """
+    Find the ophyd (dotted name) object associated with the given ophyd name.
+
+    PARAMETERS
+
+    oname
+        *str* :
+        ophyd name to search
+    force_rebuild
+        *bool* :
+        If ``True``, rebuild the internal registry that maps
+        ophyd names to ophyd objects.
+    ns
+        *dict* or `None` :
+        Namespace dictionary of Python objects.
+
+    RETURNS
+
+    str or ``None``:
+        Name of the ophyd object.
+
+    EXAMPLE::
+
+        In [45]: findpv("adsimdet_cam_acquire")
```

Comment:
typo? should it be `findname`?

Reply:
yup, that's a copy/paste error caught in its native habitat